integration-tests/src/test/kotlin/org/stellar/anchor/platform/AnchorPlatformIntegrationTest.kt

```diff
@@ -80,15 +79,22 @@ class AnchorPlatformIntegrationTest {
     @BeforeAll
     @JvmStatic
     fun setup() {
-      platformServerContext =
-        AnchorPlatformServer.start(
-          SEP_SERVER_PORT,
-          "/",
-          mapOf("stellar.anchor.config" to "classpath:/integration-test.anchor-config.yaml"),
-          true
-        )
+      val configMap =
+        mapOf("stellar.anchor.config" to "classpath:/integration-test.anchor-config.yaml")
+
+      platformServerContext = AnchorPlatformServer.start(SEP_SERVER_PORT, "/", configMap, true)
 
-      AnchorReferenceServer.start(REFERENCE_SERVER_PORT, "/")
+      ServiceRunner.startAnchorReferenceServer()
+
+      Thread {
+          StellarObservingService.start(configMap)
+          try {
+            Thread.currentThread().join()
```

Comment:
I think this is a deadlock statement because the calling thread is waiting itself to terminate.

The following snippet never prints `there`.

```java
public class TestThread {
    public static void main(String[] args) throws InterruptedException {
        System.out.println("here");
        Thread.currentThread().join();
        System.out.println("there");
    }
}
```

Reply:
Fixed in https://github.com/stellar/java-stellar-anchor-sdk/pull/546/commits/6356c8676ff6571e8d93470e9bf1acb1ac508d02.